### ISSUE TITLE
[v2.6] Unable to remove cluster

### ISSUE THREAD

**Issue by kingsd041** [rancher/rancher#36046]
**Rancher Server Setup**
- Rancher version: v2.6.3
- Installation option (Docker install/Helm Chart): Docker install
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc):
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: v1.20.14+k3s1
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions:
Admin

   
**Describe the bug**

Import a downstream cluster. When the downstream cluster is unreachable, the downstream cluster cannot be deleted
**To Reproduce**
 
1. Launch a single node Rancher on host A
2. Launch a K3s cluster on host B
3. Import K3s into Rancher Server
4. Run the uninstall K3s script `k3s-uninstall.sh` on host B, and the Rancher UI shows that the downstream cluster is unreachable
5. Remove the imported K3s cluster on Rancher UI Cluster Management
**Result**
After removing the cluster from Cluster Management, return to the home page. The cluster still exists

**Expected Result**

**Screenshots**

![image](https://user-images.githubusercontent.com/19684717/148318017-78beb0b0-1c3d-47c6-ba28-86b5055543cd.png)

![image](https://user-images.githubusercontent.com/19684717/148317988-bc0a47a5-e82c-42a9-a144-0c6765c611d8.png)

**Additional context**

This downstream cluster cannot be deleted through the local cluster：
![image](https://user-images.githubusercontent.com/19684717/148318436-ed84ebcb-3e26-4d82-8939-0e6baaf8510b.png)

**Comment by jgreat** [comment]
This is also a problem for imported clusters created by terrafrom `rancher2_cluster` resource.  

My process is to create k3s clusters and import them into rancher for management.  If the cluster doesn't have any nodes, it seems like you can't delete it.

In order to remove the entry I had to go into the `local` cluster and edit the cluster resources to remove the finalizes, which allowed the cluster to delete. Finally delete the cluster namespace.

```
root@rancher000001:~# kubectl patch Cluster.management.cattle.io c-x85jt -p '{"metadata":{"finalizers":null}}' --type=merge
cluster.management.cattle.io/c-x85jt patched

root@rancher000001:~# kubectl get Cluster.management.cattle.io  c-x85jt
Error from server (NotFound): clusters.management.cattle.io "c-x85jt" not found

root@rancher000001:~# kubectl delete namespace c-x85jt
namespace "c-x85jt" deleted
```

**Comment by jgreat** [comment]
Still an issue.

**Comment by dsmithbauer** [comment]
Agreed. Since Rancher versions > 2.6.0 I have had issues deleting clusters with the rancher2_cluster Terraform provider, and I still am having issues on 2.6.3-patch2.

Terraform will keep waiting/polling for the resource to be deleted, but the cluster.cattle.management.io resource still has one and only one finalizer left. If I remove the last finalizer and replace with `[]`, the resource is removed and Terraform is informed that the cluster has been fully deleted from Rancher as long as it hasn't timed out yet.
```
  finalizers:                                                        
  - controller.cattle.io/cluster-agent-controller-cleanup      
  generateName: c- 
```

We have tried running newer versions of the Rancher2 Terraform Provider (currently on 1.22.2) and still this is an issue.  This is a real pain for our automated DevOps pipelines.  I've read the release notes eagerly for several versions/patches of Rancher and hoped the issues addressing "stuck clusters" (https://github.com/rancher/rancher/issues/35180) deleting would fix the bug that was introduced after 2.6.0, but none of them have.

Any help prioritizing this fix would be greatly appreciated!

> This is also a problem for imported clusters created by terrafrom `rancher2_cluster` resource.
> 
> My process is to create k3s clusters and import them into rancher for management. If the cluster doesn't have any nodes, it seems like you can't delete it.
> 
> In order to remove the entry I had to go into the `local` cluster and edit the cluster resources to remove the finalizes, which allowed the cluster to delete. Finally delete the cluster namespace.
> 
> ```
> root@rancher000001:~# kubectl patch Cluster.management.cattle.io c-x85jt -p '{"metadata":{"finalizers":null}}' --type=merge
> cluster.management.cattle.io/c-x85jt patched
> 
> root@rancher000001:~# kubectl get Cluster.management.cattle.io  c-x85jt
> Error from server (NotFound): clusters.management.cattle.io "c-x85jt" not found
> 
> root@rancher000001:~# kubectl delete namespace c-x85jt
> namespace "c-x85jt" deleted
> ```
